.7z archive: `panic: runtime error: index out of range [18155] with length 328`

Please review the [Community Note](https://github.com/trufflesecurity/trufflehog/blob/main/.github/community_note.md) before submitting

### TruffleHog Version

The latest version built locally.

### Trace Output

```
🐷🔑🐷  TruffleHog. Unearth your secrets. 🐷🔑🐷

2023-10-29T09:58:39-04:00       info-0  trufflehog      handling binary file    {"job_id": 1, "source_manager_worker_id": "HhBjw", "source_type": "SOURCE_TYPE_GIT", "source_name": "trufflehog - git", "path": "ubuntu-archive-keyring.gpg", "repo": "https://github.com/ValveSoftware/steam-runtime.git", "commit": "2cb0efbbcb88818818e42e2b7bb654f69103ddb3"}
2023-10-29T09:58:39-04:00       info-0  trufflehog      handling binary file    {"job_id": 1, "source_manager_worker_id": "HhBjw", "source_type": "SOURCE_TYPE_GIT", "source_name": "trufflehog - git", "path": "packages/source/libsdl2/libsdl2_2.0.3.tar.gz", "repo": "https://github.com/ValveSoftware/steam-runtime.git", "commit": "35b3f126adfc7acbc6292616237f2ec678a626cf"}
2023-10-29T09:58:39-04:00       info-0  trufflehog      handling binary file    {"job_id": 1, "source_manager_worker_id": "HhBjw", "source_type": "SOURCE_TYPE_GIT", "source_name": "trufflehog - git", "path": "crosstool/ct-ng-win/x86_64-4.8.2-release-win32-sjlj-rt_v3-rev2.7z", "repo": "https://github.com/ValveSoftware/steam-runtime.git", "commit": "9e69aaf3b3a773626242beea9897e7bb85441d43"}
panic: runtime error: index out of range [18155] with length 328

goroutine 8504 [running]:
github.com/bodgit/sevenzip.readFilesInfo({0x7fdd20343390?, 0xc0016140c0})
        /home/user/go/pkg/mod/github.com/bodgit/sevenzip@v1.3.0/types.go:707 +0x650
github.com/bodgit/sevenzip.readHeader({0x7fdd20343390, 0xc0016140c0})
        /home/user/go/pkg/mod/github.com/bodgit/sevenzip@v1.3.0/types.go:811 +0xdc
github.com/bodgit/sevenzip.readEncodedHeader({0x7fdd20343390, 0xc0016140c0})
        /home/user/go/pkg/mod/github.com/bodgit/sevenzip@v1.3.0/types.go:849 +0x9a
github.com/bodgit/sevenzip.(*Reader).init(0xc000b72a20, {0x7fdd20343138?, 0xc0042260c0}, 0x38477bc)
        /home/user/go/pkg/mod/github.com/bodgit/sevenzip@v1.3.0/reader.go:356 +0x867
github.com/bodgit/sevenzip.NewReaderWithPassword({0x7fdd20343138, 0xc0042260c0}, 0x38477bc, {0x0, 0x0})
        /home/user/go/pkg/mod/github.com/bodgit/sevenzip@v1.3.0/reader.go:242 +0x93
github.com/mholt/archiver/v4.SevenZip.Extract({0x0?, {0x0?, 0x0?}}, {0x3bfdc68, 0xc004226660}, {0x3bcc340?, 0xc0042260c0?}, {0x0, 0x0, 0x0}, ...)
        /home/user/go/pkg/mod/github.com/mholt/archiver/v4@v4.0.0-alpha.8/7z.go:76 +0x12b
github.com/trufflesecurity/trufflehog/v3/pkg/handlers.(*Archive).openArchive(0xc000116240, {0x3bfdd10, 0xc0012fa000}, 0x0, {0x3bcc340, 0xc0042260c0}, 0xc00e4560c0)
        /home/user/dev/github.com/trufflesecurity/trufflehog/pkg/handlers/archive.go:114 +0x2b3
github.com/trufflesecurity/trufflehog/v3/pkg/handlers.(*Archive).FromFile.func1()
        /home/user/dev/github.com/trufflesecurity/trufflehog/pkg/handlers/archive.go:76 +0x1af
created by github.com/trufflesecurity/trufflehog/v3/pkg/handlers.(*Archive).FromFile in goroutine 151
        /home/user/dev/github.com/trufflesecurity/trufflehog/pkg/handlers/archive.go:71 +0xe5
```

### Expected Behavior

The archive is decoded successfully.

### Actual Behavior

A runtime error is thrown.

### Steps to Reproduce

Attempt to scan: https://github.com/ValveSoftware/steam-runtime

The problematic file appears to be: https://github.com/ValveSoftware/steam-runtime/blob/9e69aaf3b3a773626242beea9897e7bb8544/crosstool/ct-ng-win/x86_64-4.8.2-release-win32-sjlj-rt_v3-rev2.7z

## Environment
N/A

## Additional Context

This is likely a bug with https://github.com/bodgit/sevenzip. I will cross-post this there soon.

### References

N/A

--- a/go.mod
+++ b/go.mod
@@ -29,7 +29,7 @@ require (
 	github.com/felixge/fgprof v0.9.3
 	github.com/getsentry/sentry-go v0.25.0
 	github.com/go-errors/errors v1.5.1
-	github.com/go-git/go-git/v5 v5.8.1
+	github.com/go-git/go-git/v5 v5.8.0
 	github.com/go-ldap/ldap/v3 v3.4.6
 	github.com/go-logr/logr v1.3.0
 	github.com/go-logr/zapr v1.2.4
@@ -50,7 +50,7 @@ require (
 	github.com/joho/godotenv v1.5.1
 	github.com/jpillora/overseer v1.1.6
 	github.com/kylelemons/godebug v1.1.0
-	github.com/launchdarkly/go-server-sdk/v6 v6.1.1
+	github.com/launchdarkly/go-server-sdk/v6 v6.0.0-alpha.pub.12
 	github.com/lib/pq v1.10.9
 	github.com/lrstanley/bubblezone v0.0.0-20230911164824-e3824f1adde9
 	github.com/marusama/semaphore/v2 v2.5.0
@@ -65,7 +65,7 @@ require (
 	github.com/prometheus/client_golang v1.17.0
 	github.com/rabbitmq/amqp091-go v1.9.0
 	github.com/sergi/go-diff v1.3.1
-	github.com/snowflakedb/gosnowflake v1.6.25
+	github.com/snowflakedb/gosnowflake v1.6.20
 	github.com/stretchr/testify v1.8.4
 	github.com/tailscale/depaware v0.0.0-20210622194025-720c4b409502
 	github.com/trufflesecurity/disk-buffer-reader v0.1.9
@@ -114,7 +114,6 @@ require (
 	github.com/alecthomas/template v0.0.0-20190718012654-fb15b899a751 // indirect
 	github.com/alecthomas/units v0.0.0-20211218093645-b94a6e3cc137 // indirect
 	github.com/andybalholm/brotli v1.0.5 // indirect
-	github.com/apache/arrow/go/v12 v12.0.1 // indirect
 	github.com/apache/thrift v0.16.0 // indirect
 	github.com/atotto/clipboard v0.1.4 // indirect
 	github.com/aws/aws-sdk-go-v2 v1.17.7 // indirect
@@ -134,9 +133,9 @@ require (
 	github.com/aymerick/douceur v0.2.0 // indirect
 	github.com/benbjohnson/clock v1.1.0 // indirect
 	github.com/beorn7/perks v1.0.1 // indirect
-	github.com/bodgit/plumbing v1.2.0 // indirect
-	github.com/bodgit/sevenzip v1.3.0 // indirect
-	github.com/bodgit/windows v1.0.0 // indirect
+	github.com/bodgit/plumbing v1.3.0 // indirect
+	github.com/bodgit/sevenzip v1.4.3 // indirect
+	github.com/bodgit/windows v1.0.1 // indirect
 	github.com/cespare/xxhash/v2 v2.2.0 // indirect
 	github.com/cloudflare/circl v1.3.3 // indirect
 	github.com/connesc/cipherio v0.2.1 // indirect
@@ -189,7 +188,7 @@ require (
 	github.com/jpillora/s3 v1.1.4 // indirect
 	github.com/kevinburke/ssh_config v1.2.0 // indirect
 	github.com/klauspost/asmfmt v1.3.2 // indirect
-	github.com/klauspost/compress v1.16.5 // indirect
+	github.com/klauspost/compress v1.16.6 // indirect
 	github.com/klauspost/cpuid/v2 v2.2.3 // indirect
 	github.com/klauspost/pgzip v1.2.5 // indirect
 	github.com/launchdarkly/ccache v1.1.0 // indirect
@@ -220,7 +219,7 @@ require (
 	github.com/onsi/gomega v1.23.0 // indirect
 	github.com/opencontainers/go-digest v1.0.0 // indirect
 	github.com/opencontainers/image-spec v1.1.0-rc3 // indirect
-	github.com/pierrec/lz4/v4 v4.1.17 // indirect
+	github.com/pierrec/lz4/v4 v4.1.18 // indirect
 	github.com/pjbgf/sha1cd v0.3.0 // indirect
 	github.com/pkg/browser v0.0.0-20210911075715-681adbf594b8 // indirect
 	github.com/pkg/diff v0.0.0-20200914180035-5b29258ca4f7 // indirect
@@ -234,7 +233,7 @@ require (
 	github.com/sirupsen/logrus v1.9.1 // indirect
 	github.com/skeema/knownhosts v1.2.0 // indirect
 	github.com/therootcompany/xz v1.0.1 // indirect
-	github.com/ulikunitz/xz v0.5.10 // indirect
+	github.com/ulikunitz/xz v0.5.11 // indirect
 	github.com/vbatts/tar-split v0.11.3 // indirect
 	github.com/xanzy/ssh-agent v0.3.3 // indirect
 	github.com/xdg-go/pbkdf2 v1.0.0 // indirect

